### PR TITLE
Use toMatchSnapshot for expectMetricsJson

### PR DESCRIPTION
**Pull request by edi9999**

### CODE REVIEW

**Comment by edi9999** [comment]
Fixes https://github.com/stryker-mutator/stryker-js/issues/3359

**Comment by edi9999** [comment]
I've fixed the above error by installing enzyme-to-json in the e2e directory.

File: e2e/helpers.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import { promises as fsPromises } from 'fs';
 
 import { mutationTestReportSchema } from '@stryker-mutator/api/report';
-import { expect } from 'chai';
+import chai from 'chai';
 import path from 'path';
 import { calculateMutationTestMetrics, MetricsResult, Metrics } from 'mutation-testing-metrics';
 import { execSync, ExecException } from 'child_process';
+import { jestSnapshotPlugin } from "mocha-chai-jest-snapshot";
```
Comment (nicojs) [inline e2e/helpers.ts:8]: Why the `mocha-chai-jest-snapshot` plugin vs the `chai-jest-snapshot` plugin?

File: packages/instrumenter/test/unit/mutators/conditional-expression-mutator.spec.ts [hunk 2]
```diff
@@ -70,6 +70,50 @@ describe(sut.name, () => {
     );
   });
 
+  it('should not mutate (a || b) condition to (a || true)', () => {
```
Comment (nicojs) [inline packages/instrumenter/test/unit/mutators/conditional-expression-mutator.spec.ts:73]: This shouldn't be in this PR, probably need to merge master into this branch.

File: e2e/helpers.ts [hunk 3]
```diff
@@ -88,7 +91,7 @@ export async function expectMetricsResult(expectedMetricsResult: Partial<Metrics
 
 export async function expectMetricsJson(expectedMetrics: Partial<Metrics>) {
```
Comment (nicojs) [inline e2e/helpers.ts:92]: I don't think the `expectedMetrics` are used anymore, right? Please remove everywhere. I also think we should rename this to `expectMetricsJsonToMatchSnapshot()`.

**Review by nicojs — CHANGES_REQUESTED** [review]
I've got some small remarks. I think (hope?) we can fix the enzyme stuff by regenerating the `package.lock.json` files, I can try if you want (after you've implemented the fixes)

File: e2e/helpers.ts (hunk 1 above)
Comment (edi9999) [reply e2e/helpers.ts]: Good catch, I don't know how I mixed them up, I've fixed this now.

File: e2e/helpers.ts (hunk 3 above)
Comment (edi9999) [reply e2e/helpers.ts]: Right, I have changed this

File: packages/instrumenter/test/unit/mutators/conditional-expression-mutator.spec.ts (hunk 2 above)
Comment (edi9999) [reply packages/instrumenter/test/unit/mutators/conditional-expression-mutator.spec.ts]: Right, I'm removing this.

**Comment by edi9999** [comment]
I don't have a clue why, but the tests are passing for me locally, but not on the CI : 

For example, for the babel-transpiling project : 

My current master generates the following :

```
-----------|---------|----------|-----------|------------|----------|---------|
File       | % score | # killed | # timeout | # survived | # no cov | # error |
-----------|---------|----------|-----------|------------|----------|---------|
All files  |   57.45 |       27 |         0 |         20 |        0 |       1 |
 Bank.js   |   70.00 |       14 |         0 |          6 |        0 |       0 |
 Casino.js |   46.15 |       12 |         0 |         14 |        0 |       1 |
 User.js   |  100.00 |        1 |         0 |          0 |        0 |       0 |
-----------|---------|----------|-----------|------------|----------|---------|
```

But the CI generates the following : 

```
-----------|---------|----------|-----------|------------|----------|---------|
File       | % score | # killed | # timeout | # survived | # no cov | # error |
-----------|---------|----------|-----------|------------|----------|---------|
All files  |   55.56 |       25 |         0 |         20 |        0 |       1 |
 Bank.js   |   70.00 |       14 |         0 |          6 |        0 |       0 |
 Casino.js |   41.67 |       10 |         0 |         14 |        0 |       1 |
 User.js   |  100.00 |        1 |         0 |          0 |        0 |       0 |
-----------|---------|----------|-----------|------------|----------|---------|
```

I don't understand what is happening, it seems my local code has something wrong.

**Comment by nicojs** [comment]
Wow 🤯. I'll take a look when I have some time

**Comment by nicojs** [comment]
I think I know what's going on. The PR builds run on the Pull request head. Meaning after merging to master. Your changes from #3346 were probably not merged on your local branch (maybe?). Trying now after merging master and updating snaps

**Comment by edi9999** [comment]
Good catch that was the case indeed

**Comment by nicojs** [comment]
Thanks a lot @edi9999 ! 💖

**Pull request merged**